pkg/utils/resources/resources.go

```diff
@@ -51,6 +51,9 @@ func Merge(resources ...v1.ResourceList) v1.ResourceList {
 	result := make(v1.ResourceList, len(resources[0]))
 	for _, resourceList := range resources {
 		for resourceName, quantity := range resourceList {
+			if resourceName == "hugepages-2Mi" || resourceName == "hugepages-1Gi" {
```

Comment:
We had a similar ask [here](https://github.com/aws/karpenter/issues/1900) but we had to reject the idea because ignoring the resource means that we can't know the best instance type to launch your workload pod on. You can imagine that we launch an instance type which doesn't actually support the resource, or we schedule a node that doesn't have enough of the resource to support your pod. Ideally, we would be able to return the exact resources each InstanceType has through the [GetInstanceTypes](https://github.com/aws/karpenter/blob/main/pkg/cloudprovider/cloudprovider.go#L171) API

Reply:
@jonathan-innis Thanks for the review.
Regarding - 
> You can imagine that we launch an instance type which doesn't actually support the resource

That's what I tried to describe above, such resource as _hugepages_ isn't a HW based (instance type based) resource, but a SW based resource, i.e. you'll never know if your node/instance type supports the resource.

Anyway, how that behavior can be achieved? We want to migrate to Karpenter but we have to support 'hugepages' resource.

Thank you.

Reply:
It is a SW-based component, but I believe that it scales based on the size of the instance, which means it's difficult to achieve a fixed sizing that we could pass through. Is this something that should be consistent on your kernel across all instance types or just the few that you've associated with a given image and NodeTemplate?

Reply:
The second option.

Reply:
@jonathan-innis, since its a specific request for a specific instance type (DL1) which is always needed (there is no point running habanlabs sw pod on DL1 without hugepages - it will not work), What if we add this condition/ignore case only for DL1 instance type will that be, Ok?

Reply:
Is it safe to assume hugepages to be some large number in instance type provider? Is this something that we should specify in the AWSNodeTemplate or Provisioner.spec.resources?

Reply:
@ellistarn generally speaking - Provisioner.spec.resources